Cython/Utility/CConvert.pyx

```diff
@@ -18,7 +18,7 @@ cdef {{struct_type}} {{funcname}}(obj) except *:
         value = obj['{{member.name}}']
     except KeyError:
         raise ValueError("No value specified for struct attribute '{{member.name}}'")
-    result.{{member.cname}} = value
+    result.{{member.name}} = value
```

Comment:
What?

Reply:
This is generating Cython code so I believe it's correct to use the name, rather than the cname (although the vast majority of the time they're the same). Here's a non-unicode example of something that works now but didn't before:

```
cdef struct S:
    int x "not_x"
def f(d):
    """
    >>> f({ 'x': 1 })
    1
    """
    cdef S s = d
    print(s.x)
```

Although really this is related to the struct support (which is getting moved out of this PR) I'm going to leave this bit in because it fixes existing bugs.

Reply:
Ah, right. Yes, that's correct. Did you added this example as a test below?

Reply:
Have done now. Also added a test for conversion the other way (which wasn't broken, but if you're testing one way it seems worth testing the other)